src/admin/components/ResourceTabs/PolygonReviewTab/components/PolygonStatus/StatusDisplay .tsx

```diff
@@ -0,0 +1,249 @@
+    title: "Draft",
+    status: "draft",
+    value: 1
+  },
+  {
+    title: "Awaiting Approval",
+    status: "awaiting-approval",
+    value: 2
+  },
+  {
+    title: "Needs More Information",
+    status: "needs-more-information",
+    value: 3
+  },
+  {
+    title: "Planting in Progress",
+    status: "planting-in-progress",
+    value: 4
+  },
+  {
+    title: "Approved",
+    status: "approved",
+    value: 5
+  }
+];
+const menuProjectOptions = [
+  {
+    title: "Draft",
+    status: "draft",
+    value: 1
+  },
+  {
+    title: "Awaiting Approval",
+    status: "awaiting-approval",
+    value: 2
+  },
+  {
+    title: "Needs More Information",
+    status: "needs-more-information",
+    value: 3
+  },
+  {
+    title: "Approved",
+    status: "approved",
+    value: 4
+  }
+];
+
+export interface StatusProps {
+  titleStatus: "Site" | "Project" | "Polygon";
+  mutate?: any;
+  record?: any;
+  refresh?: any;
+  name: any;
+  refetchPolygon?: any;
+  setSelectedPolygon?: any;
+}
+
+const menuOptionsMap = {
+  Polygon: menuPolygonOptions,
+  Site: menuSiteOptions,
+  Project: menuProjectOptions
+};
+
+const DescriptionStatusMap = {
+  Polygon: "Are you sure you want to change the polygon status to",
+  Site: "Are you sure you want to change the site status to",
+  Project: "Are you sure you want to change the project status to"
+};
+
+const DescriptionRequestMap = {
+  Polygon: "Provide an explanation for your change request for the polygon",
+  Site: "Provide an explanation for your change request for the site",
+  Project: "Provide an explanation for your change request for the project"
+};
+
+const StatusDisplay = ({ titleStatus = "Polygon", mutate, refresh, name, record, setSelectedPolygon }: StatusProps) => {
+  const { refetch: reloadEntity } = useShowContext();
+  const [notificationStatus, setNotificationStatus] = useState<{
+    open: boolean;
+    message: string;
+    type: "success" | "error" | "warning";
+    title: string;
+  }>({
+    open: false,
+    message: "",
+    type: "success",
+    title: "Success!"
+  });
+
+  const { openModal, closeModal } = useModalContext();
+  const contentStatus = (
+    <Text variant="text-12-light" as="p" className="text-center">
+      {DescriptionStatusMap[titleStatus]} <b style={{ fontSize: "inherit" }}>{name}</b>?
+    </Text>
+  );
+  const contentRequest = (
+    <Text variant="text-12-light" as="p" className="text-center">
+      {DescriptionRequestMap[titleStatus]} <b style={{ fontSize: "inherit" }}>{name}</b>?
+    </Text>
+  );
+
+  const openFormModalHandlerStatus = () => {
+    openModal(
+      <ModalConfirm
+        title={`${titleStatus} Status Change`}
+        commentArea
+        menuLabel={""}
+        menu={menuOptionsMap[titleStatus]}
+        onClose={closeModal}
+        content={contentStatus}
+        onConfirm={async (text: any, opt) => {
+          const option = menuOptionsMap[titleStatus].find(option => option.value === opt[0]);
+          let response;
```

Comment:
Since this is never reassigned and only used inside the try block, please declare as a const on line 144

Reply:
Thanks for the feedback! I've updated the code to declare the variable as a constant within the try block, as suggested.